packages/scratch-vm/src/extensions/scratch3_spotify/index.ts

```diff
@@ -0,0 +1,492 @@
+ * @description As you can see, each block is represented as a function.
+ * In typescript, you can specify a function in either of the following ways (and which you choose is a matter of preference):
+ * - Arrow syntax: `nameOfFunction: (argument1Name: argument1Type, argument2Name: argument2Type, ...etc...) => returnType;`
+ * - 'Method' syntax: `nameOfFunction(argument1Name: argument1Type, argument2Name: argument2Type, ...etc...): returnType;`
+ *
+ * The three included functions demonstrate some of the most common types of blocks: commands, reporters, and hats.
+ * - Command functions/blocks take 0 or more arguments, and return nothing (indicated by the use of a `void` return type).
+ * - Reporter functions/blocks also take 0 or more arguments, but they must return a value (likely a `string` or `number`).
+ * - Hat functions/blocks also take 0 or more arguments, but they must return a boolean value.
+ *
+ * Feel free to delete these once you're ready to implement your own blocks.
+ *
+ * This type will be passed as the second generic argument to the Extension class that your specific extension 'extends'
+ * (see the class defintion below for more information on extending the Extension base class).
+ * @link https://www.typescriptlang.org/docs/handbook/2/functions.html Learn more about function types!
+ * @link https://www.typescriptlang.org/docs/handbook/2/objects.html Learn more about object types! (This is specifically a 'type alias')
+ * @link https://www.typescriptlang.org/docs/handbook/2/generics.html Learn more about generics!
+ */
+type Blocks = {
+    /*exampleCommand(exampleString: string, exampleNumber: number): void;
+  exampleReporter: () => number;
+  exampleHat(condition: boolean): boolean;*/
+    searchAndPlay(query: string): void;
+    // TODO uncomment when blocks are implemented
+    /*searchAndPlayWait(query: string): void;
+  playNextBeat(): void;
+  playBeat(beatNum: number): void;
+  playBeatAndWait(beatNum: number): void;*/
+    trackData(trackAttr: string): string;
+    /*currentBeat(): number;
+  everyBeat(): boolean;
+  everyBar(): boolean;
+  stopMusic(): void; */
+};
+
+/**
+ * @summary This is the class responsible for implementing the functionality of your blocks.
+ * @description You'll notice that this class `extends` (or 'inherits') from the base `Extension` class.
+ *
+ * Hover over `Extension` to get a more in depth explanation of the base class, and what it means to `extend it`.
+ */
+class Spotify extends Extension<Details, Blocks> {
+    /**
+     * @summary A field to demonstrate how Typescript Class fields work
+     * @link https://www.typescriptlang.org/docs/handbook/2/classes.html#fields
+     */
+    // Example field 1
+    //exampleField: number;
+    prevQuery: string;
+    currentArtistName: string;
+    currentTrackName: string;
+    currentAlbumName: string;
+
+    spotifyToken: AccessToken;
+
+    // FEEDBACK using JS library, not sure what type to put
```

Comment:
Depending on how things go `npm install`ing Tone.js, we might be able to source types from there (as mentioned above)